### ISSUE TITLE
UnusedPrivateMember doesn't report the correct warning count

### ISSUE THREAD

**Issue by BraisGabin** [arturbosch/detekt#1981]
## Expected Behavior
```kotlin
class Test {
    private fun unusedFunction(): Int {
        return 5
    }
}

class Test2 {
    private fun unusedFunction(): Int {
        return 5
    }
}
```

In that code I expect to have two warnings because there are two unused functions

## Observed Behavior
Only one warning is raised. The reason is becase the functions are called the same.

## Steps to Reproduce
You can check this comparison: https://github.com/arturbosch/detekt/compare/master...BraisGabin:unused-private-function

## Context
I found this issue fixing a similar issue with parameters that are called the same in different functions. That issue was easier to fix that this one.

## Your Environment
* Version of detekt used: 1.0.1
* Operating System and version: Mac

**Comment by schalkms** [comment]
This rule only flags unused functions/members/parameters in classes/interfaces/objects.
Therefore, it only checks *members* of classes/interfaces/objects.
This won't be included in this rule, since we would need to change the name and description of the rule.
Anyway, thanks for battle testing detekt while implementing the HTML report.
Your feedback is really valuable in that regard.

**Comment by BraisGabin** [comment]
Ok, I edited the snippet using just methods inside classes and the error is there. Can we reopen this?

**Comment by arturbosch** [comment]
Yep, the rule currently counts the references to each name in a hashmap.
Storing the whole path taken from top level till the variable etc could solve this problem.
e.g.: `ClassA>FunctionA>UnusedVar to KtElement` instead of just `UnusedVar to KtElement`

**Comment by sowmyav24** [comment]
Do we want to handle function overloading as part of this rule ?
Eg:
```
class Test {
    private fun unusedFunction(): Int {
        return 5
    }
   private fun unusedFunction(num: Int): Int {
        return num
    }
}
```

**Comment by BraisGabin** [comment]
I updated the one proposed by @sowmyav24 and added two more (the other one is the same but with two functions):

```
class Test {
    val value = unusedFunction()

    private fun unusedFunction(): Int {
        return 5
    }

    private fun unusedFunction(num: Int): Int {
        return num
    }
}
```

If we have two functions that are called the same and one of them is used, the other one is not reported. I think that this case is a bit worse that the first that I reported. Here you are getting 0 findings.

**Comment by schalkms** [comment]
Thanks for formulating new test cases and for investigating this issue further.

**Comment by BraisGabin** [comment]
I think that to fix this issue we need type resolver. I tried to fix it but I found a wall. If I have this functions:

```kotlin
private fun f(a: Int)
private fun f(a: String)
```

I can't know if `f("".randomExtensionFunction())` is calling the first or the second.

**Comment by schalkms** [comment]
Yes!
